mkdocs/tests/utils/utils_tests.py

```diff
@@ -321,3 +321,32 @@ def test_copy_files_without_permissions(self):
                     os.chmod(src, stat.S_IRUSR | stat.S_IWUSR)
             shutil.rmtree(src_dir)
             shutil.rmtree(dst_dir)
+
+    def test_unicode_clean_directory(self):
+        temp_dir = tempfile.mkdtemp()
+        utf8_directory = os.path.join(temp_dir, '导航'.decode('utf-8'))
```

Comment:
This isn't necessary. Up at the top of the file we have `from __future__ import unicode_literals` which ensures that all literal strings defined in in the module are Unicode strings. Its basically the same as the `u` prefix: `u'unicode string'`.

What we need t check is that `temp_dir` is Unicode. Or any other path which is not created by us.

Reply:
Thanks, now I figure out that, when user os.path.join(arg1, arg2), if both arg1 and arg2 is str or unicode, it will work, but if arg1 and arg2 are mixed of str and unicode, python will encode the str arg with system defaultencoding, which on windows is not utf-8, and it throws exceptions.   
as tempfile didn't work with utf-8 well , so I make utf-8 dir myself:  
```
temp_dir = tempfile.mkdtemp()
utf8_temp_dir = os.path.join(dec('tmp'), dec('tmp_utf8'))
```